iamrole/validate.go

```diff
@@ -5,109 +5,81 @@ import (
 	"strings"
 	"unicode"
 
+	"go.einride.tech/aip/validation"
 	"go.einride.tech/iam/iampermission"
 	"google.golang.org/genproto/googleapis/iam/admin/v1"
-	"google.golang.org/genproto/googleapis/rpc/errdetails"
 )
 
 // Validate checks that an IAM role is valid.
-func Validate(role *admin.Role) *errdetails.BadRequest {
-	var result errdetails.BadRequest
-	result.FieldViolations = appendNameViolations(result.FieldViolations, role.GetName())
-	result.FieldViolations = appendTitleViolations(result.FieldViolations, role.GetTitle())
-	result.FieldViolations = appendDescriptionViolations(result.FieldViolations, role.GetTitle())
-	result.FieldViolations = appendIncludedPermissionsViolations(result.FieldViolations, role.GetIncludedPermissions())
-	if len(result.FieldViolations) > 0 {
-		return &result
-	}
-	return nil
+func Validate(role *admin.Role) error {
+	var result validation.MessageValidator
+	addNameViolations(&result, role.GetName())
+	addTitleViolations(&result, role.GetTitle())
+	addDescriptionViolations(&result, role.GetTitle())
+	addIncludedPermissionsViolations(&result, role.GetIncludedPermissions())
+	return result.Err()
 }
 
-func appendNameViolations(
-	fieldViolations []*errdetails.BadRequest_FieldViolation,
-	name string,
-) []*errdetails.BadRequest_FieldViolation {
+func addNameViolations(result *validation.MessageValidator, name string) {
```

Comment:
`checkNameViolations`? Because it does not add anything if there are not violations.

Reply:
I see! The naming is meant to match up with the `AddFieldViolations` API

Reply:
Not exported, so I'm fine with either, but prefer `check[...]`